internal/graph/check.go

```diff
@@ -659,11 +667,15 @@ func (c *LocalChecker) buildCheckAssociatedObjects(req *ResolveCheckRequest, obj
 			})
 		}
 
+		objectIDsArrayOfStrings := make([]string, objectIDs.Size())
+		for i, objectID := range objectIDs.Values() {
+			objectIDsArrayOfStrings[i] = objectID.(string)
+		}
 		i, err := ds.ReadStartingWithUser(ctx, storeID, storage.ReadStartingWithUserFilter{
 			ObjectType: objectType,
 			Relation:   relation,
 			UserFilter: userFilter,
-			ObjectIDs:  maps.Keys(objectIDs),
+			ObjectIDs:  objectIDsArrayOfStrings,
```

Comment:
Why convert it into a slice? We can let the storage implementation decide how they want to use/implement it.

Reply:
I wasn't sure whether we wanted to break the interface. I can do that